src/python/pants/backend/python/sandboxed_interpreter.py

```diff
@@ -0,0 +1,64 @@
+# coding=utf-8
+# Copyright 2018 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+import os
+
+from pex.interpreter import PythonInterpreter
+
+from pants.util.memo import memoized_method
+
+
+class SandboxedInterpreter(PythonInterpreter):
+
+  class ToolchainLocationError(Exception):
+    def __init__(self, dir_path):
+      msg = "path '{}' does not exist or is not a directory".format(dir_path)
+      super(ToolchainLocationError, self).__init__(msg)
+
+  class BaseInterpreterError(Exception): pass
+
+  # using another PythonInterpreter to populate the superclass constructor args
+  def __init__(self, llvm_base_dir, base_interp):
+
+    if not os.path.isdir(llvm_base_dir):
+      raise ToolchainLocationError(llvm_base_dir)
```

Comment:
Would it be possible for this to cause reproducibility issues across devel and prod environments?

Reply:
AFAIK, this failure would mean something other than "we don't have a toolchain for your OS". It would mean something like: "we tried downloading it and partially succeeded, but then someone maybe deleted some of it".

Reply:
The existing cpp contrib does the fallback you describe, and we're replacing it because controlling the toolchain allows us to reliably orchestrate complex things like native code in setup.py-based projects -- I think relinquishing that would be defeating the purpose. The reason packaging the toolchain (separately from its consumption in Pants) is nontrivial is because it needs to work across all of Pants's supported platforms reliably, so that we don't need to provide a fallback, in my understanding.

Reply:
See below comment in the main PR thread.

Reply:
And yes, @stuhood is correct!